language/stdlib/modules/RecoveryAddress.move

```diff
@@ -77,33 +79,33 @@ module RecoveryAddress {
             i = i + 1
         };
         spec {
-            assert i == len + 1;
+            // TODO(shb): Prover thinks this doesn't hold. Is it because of the return in the loop?
+            // How is tis spec block interpreted?
+            //assert i == len;
```

Comment:
I think you also need to change line 68 to `assert i <= len;`. Prover relies on the loop invariants to prove line 84.

Reply:
Thanks, that did it!